### PR TITLE
[fix][admin] Fix examine messages if total message is zero

### PR DESCRIPTION
**Pull request by AnonHxy**
### Motivation

If the total messages is zero, we should throw exception fast, insteading  throwing exception when reading from the ledger.
### Modifications

If the total messages is zero, throw exception

### Verifying this change

- [x] Make sure that the change passes the CI checks.
- org.apache.pulsar.broker.admin.PersistentTopicsTest#testExamineMessage will cover this test

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/AnonHxy/pulsar/pull/36

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -3045,6 +3045,10 @@ protected CompletableFuture<Response> internalExamineMessageAsync(String initial
                     try {
                         PersistentTopic persistentTopic = (PersistentTopic) topic;
                         long totalMessage = persistentTopic.getNumberOfEntries();
+                        if(totalMessage <= 0) {
+                            throw new RestException(Status.INTERNAL_SERVER_ERROR,
```
Comment (lifepuzzlefun) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3049]: i think we can change http status code to 422 Unprocessable Content or 412 Precondition Failed

Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: Great. 412 accepted

Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: @lifepuzzlefun PTAL

Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: @Technoboy- @codelipenghui @lordcheng10  @HQebupt PTAL

**Comment by lifepuzzlefun** [comment]
@AnonHxy hi, after check the intention of this admin interface. i think we can return an meaningful response directly without return exception to the caller. like if there is no message here, any check should return equal to latest or earliest WDYT

**Comment by AnonHxy** [comment]
> @AnonHxy hi, after check the intention of this admin interface. i think we can return a meaningful response directly without return exception to the caller. like if there is no message here, any check should return equal to latest or earliest, WDYT?

Do you mean that we return an empty response, like `Response.ok().build()`?  Maybe this will confuse the users I think. @lifepuzzlefun

**Comment by lifepuzzlefun** [comment]
> > @AnonHxy hi, after check the intention of this admin interface. i think we can return a meaningful response directly without return exception to the caller. like if there is no message here, any check should return equal to latest or earliest, WDYT?
> 
> Do you mean that we return an empty response, like `Response.ok().build()`? Maybe this will confuse the users I think. @lifepuzzlefun

I see, i misunderstand the intention. : - )

**Review by lifepuzzlefun — APPROVED** [review]

**Review by StevenLuMT — APPROVED** [review]
LGTM

**Review by HQebupt — APPROVED** [review]
Good catch!

**Pull request merged**